src/win/winsock.c

```diff
@@ -559,3 +559,32 @@ int WSAAPI uv_msafd_poll(SOCKET socket, AFD_POLL_INFO* info_in,
     return SOCKET_ERROR;
   }
 }
+
+int uv__convert_to_localhost_if_unspecified(const struct sockaddr* addr,
```

Comment:
This whole method could be cleaned up using a union and switch similar to the code found [here](https://github.com/libuv/libuv/blob/8a9585232e9df227bcaf1263748d8997ec5d1077/src/unix/udp.c#L348-L352)